server/config/auth.js

```diff
@@ -89,20 +122,46 @@ passport.deserializeUser((id, done) => {
     .catch(err => done(null, false, { message: err.message }));
 });
 
-const setDemoUser = passport.authenticate('local', {
-  failureRedirect: '/'
-});
+const authenticate = (req, resp, next, type) => {
+  passport.authenticate(type, (err, user) => {
+    if (err) {
+      return next(err);
+    }
+
+    if (!user) {
+      return resp.sendStatus(401);
+    }
+
+    req.login(user, err => {
+      if (err) {
+        return next(err);
+      }
+
+      next();
```

Comment:
is `next()` same as `next(falsyValue)`? if yes then one line can be used: `err => next(err)`

Reply:
This next() functionality is a little bit complicated for me. If an error is 

> ... returned from asynchronous functions invoked by route handlers and middleware, you must pass them to the next() function, where Express will catch and process them.
> If you pass anything to the next() function (except the string 'route'), Express regards the current request as being an error and will skip any remaining non-error handling routing and middleware functions.

So I think, that if I pass anything to the 'next()' function,  express.js will treat it as an error. So I if I would like to go to the next function in a chain of middlewares I should call next() function with no params.